2020-02-Spring/2_linked_lists/README.md

```diff
@@ -94,31 +133,64 @@ Go to [Solution](#s3)   [Top](#top)
 <!-- Don't remove -->
 <a name="s1"/>
 
-### 1. SOLUTION 1 TODO :bug:
+### 1. Remove Nth Node From End of List
 
-Source: TODO :bug:
+Source: [LeetCode](https://leetcode.com/problems/remove-nth-node-from-end-of-list/solution/)
+
+#### Optimal Solution
 
-#### Naive/Simple Solution
 <details>
 <summary>Click to see solution</summary>
 
-TODO put your solution here :bug:
+### One pass algorithm
 
-</details>
+**Algorithm:**
 
+Instead of using one pointer, we could use two pointers. The first pointer advances the list by n+1 steps from the beginning, while the second pointer starts from the beginning of the list. Now, both pointers are exactly separated by n nodes apart. We maintain this constant gap by advancing both pointers together until the first pointer arrives past the last node. The second pointer will be pointing at the n-th node counting from the last. We relink the next pointer of the node referenced by the second pointer to point to the node's next next node:
```

Comment:
Is using one pointer the naive solution for this problem?

Reply:
The naive solution could be anything that might not use the two pointers approach. Let me rephrase this so it's not confusing. Thanks.